src/bool.rs

```diff
@@ -0,0 +1,132 @@
+//! Functions for safe transmutation to `bool`.
+//! Transmuting to `bool' is not undefined behavior if the transmuted value is either 0 or 1.
+//! These functions will return an error if the integer value behind the `bool` value is neither one.
+
+use self::super::{Error, ErrorReason, guarded_transmute_vec_permissive, guarded_transmute_many_permissive, guarded_transmute_many_pedantic,
+                  guarded_transmute_vec_pedantic};
+
+use std::mem::align_of;
+
+/// Makes sure that the bytes represent a sequence of valid boolean values. It is done
+/// this way because the language does not guarantee that `bool` is 1 byte sized.
+#[cfg(target_endian = "little")]
+#[inline]
+fn bytes_are_bool(v: &[u8]) -> bool {
+    let sizeof_bool: usize = align_of::<bool>();
+    for c in v.chunks(sizeof_bool).filter(|c| c.len() == sizeof_bool) {
+        if c[0] > 1 || c[1..].iter().any(|x| *x != 0) {
+            return false;
+        }
+    }
+    true
+}
+
+/// Makes sure that the bytes represent a sequence of valid boolean values. It is done
+/// this way because the language does not guarantee that `bool` is 1 byte sized.
+#[cfg(target_endian = "big")]
+#[inline]
+fn bytes_are_bool(v: &[u8]) -> bool {
+    let sizeof_bool: usize = align_of::<bool>();
+    for c in v.chunks(sizeof_bool).filter(|c| c.len() == sizeof_bool)  {
```

Comment:
What is `rustfmt`, amirite?

Reply:
Ain't nobody got time fo' that. `$ rustfmt src/bool.rs`